db/compaction/compaction_picker.cc

```diff
@@ -39,7 +39,14 @@ bool FindIntraL0Compaction(const std::vector<FileMetaData*>& level_files,
                            size_t min_files_to_compact,
                            uint64_t max_compact_bytes_per_del_file,
                            uint64_t max_compaction_bytes,
-                           CompactionInputFiles* comp_inputs) {
+                           CompactionInputFiles* comp_inputs,
+                           SequenceNumber oldest_mem_seqno) {
+
+  // Do not pick ingested file when there is at least one memtable not flushed which of seqno is overlap with the sst.
```

Comment:
Do you think it would work to pick the span of files for intra-L0 starting at the smallest index `i` for which `level_files[i]->fd.largest_seqno < oldest_mem_seqno`?

Reply:
That's a nicer phrasing of a suggestion I had in a comment. I don't see anything wrong with this right now, but I'd like to think harder about it.

Reply:
Sorry I loaded the page before you reviewed and didn't notice the repetition. I haven't thought of any problems with it. I also plan to experiment with relaxing the restriction more generally on intra-L0 picking from the newest file as it might be related to our imports getting stuck.

Reply:
I would try it.

Reply:
This approach looks good to me.

Reply:
@ajkr It seems to me it could still order the output to be newer than `L0_(i-1)` if `L0_i` is an ingested file and `L0_i->largest_seqno < oldest_mem_seqno`.